### ISSUE TITLE
wpautop applied to FSE template parts

### ISSUE THREAD

**Issue by hawkmusician** [WordPress/gutenberg#27857]
**Describe the bug**
`wpautop` seems to be applying on template parts in FSE (using dynamic blocks, image tags are wrapped with <p> and <br> is added automatically), which completely destroys the layout on the front end.

`remove_filter( 'the_content', 'wpautop' )` not doing anything either.

**Comment by hawkmusician** [comment]
To add to this, it's in the `render_block_core_template_part` function that `$content = wpautop( $content );` is run which adds automatic paragraphs and breaks to blocks when not wanted. 

Is this intended?

**Comment by pradeep910** [comment]
This wpautop is causing issue in AMP pages too. For example liveblog `amp-live-list` doesn't work. https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/template-part/index.php#L62

![Screenshot 2021-01-23 at 5 55 06 PM](https://user-images.githubusercontent.com/1139853/105578247-accbf700-5da4-11eb-95a9-ddb94e3a2856.png)

**Comment by Sandstromer** [comment]
wpautop in template parts is also breaking the alignment styling that was added in  #29335

When using "Inherit default layout" on a block _within_ the template part

```
<style>			.wp-container-608186cd787a8 > * {
				max-width: 610px;
				margin-left: auto;
				margin-right: auto;
			}</p>
<p>			.wp-container-608186cd787a8 > .alignwide {
				max-width: 1240px;
			}</p>
<p>			.wp-container-608186cd787a8 .alignfull {
				max-width: none;
			}
</style>
```

**Comment by Sandstromer** [comment]
Looks like fixed in #30552

**Comment by skorasaurus** [comment]
Closing this since it's been fixed in #30552 ; please reopen if it's still applicable.
